app/pods/home/editor/template.hbs

```diff
@@ -2,20 +2,115 @@
 
 <div class="mainBody">
   <div class="component">
-    {{playground-button buttonAction=(action "toggleSave") text="Save"}}
+    {{playground-button buttonAction=(action "toggleSave") text=saveButtonText}}
+    {{playground-button buttonAction=(action "enableComponents") text=componentButtonText}}
   </div>
 
   <div class="component">
-    {{map-component
+    {{#map-component
         class="leaflet-container"
-        latVal="-25.3444"
-        lngVal="131.0369"
-        zoomVal="1"
+        latVal=latitude
+        lngVal=longitude
+        zoomVal=zoom
         geojson=geojson
         points=points
         saveEvent=saveEvent
         drawEnabled=true
-        destination=destinationChoice}}
+        destination=destinationChoice
+        mapInstance=mapInstance
+        imageOverlays=imageOverlays
+        videoOverlays=videoOverlays
+        componentOverlays=componentOverlays}}
+      {{#if (and mapInstance componentsEnabled)}}
+        {{!-- Display group to capture a unique layer for display boxes/points --}}
+        {{#map-component/display-group
+            mapInstance=mapInstance
+            layerGroup=layerGroup
+            layerName=layerGroupName
+            componentOverlays=componentOverlays}}
+          {{#if layerGroup}}
+            {{!-- Display point to render child components on the map at the given lat/lng --}}
+            {{#map-component/display-point
+                layerId=firstLayerId
+                lat=firstLayerLatitude
+                lng=firstLayerLongitude
+                mapInstance=mapInstance
+                layerGroup=layerGroup
+                layerReady=layerReady}}
```

Comment:
how does this layerReady work? was display-point meant to yield layerReady so you could use it in a block component here? eg

```
{{#map-component/display-point 
    ...
as |dp|}}

   {{#if dp.layerReady}}
      ...
   {{/if}}
{{/map-component/display-point }}
```

Reply:
That was my intention, I'll update to use this cleaner approach